File: bilibili-recommend-ui/app/src/main/java/com/bilibilirecommend/main/widget/adapter/RecommendActivityAdapter.java
package com.bilibilirecommend.main.widget.adapter;



import com.bilibilirecommend.R;
import com.bilibilirecommend.base.BaseListRecyclerAdapter;
import com.bilibilirecommend.base.SuperViewHolder;
import com.bilibilirecommend.main.model.RecommendModel;
import com.bilibilirecommend.utils.ImageLoaderUtils;

import java.util.List;

/**
 * by y on 2016/9/19.
 */
public class RecommendActivityAdapter extends BaseListRecyclerAdapter<RecommendModel.ResultBean.BodyBean> {


    public RecommendActivityAdapter(List<RecommendModel.ResultBean.BodyBean> mDatas) {
        super(mDatas);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.item_recommend_activity_item;
    }

    @Override
    protected void onBind(SuperViewHolder viewHolder, int position, RecommendModel.ResultBean.BodyBean mDatas) {
        ImageLoaderUtils.display(viewHolder.getImageView(R.id.iv_title_page), mDatas.getCover());
        viewHolder.setTextView(R.id.tv_title, mDatas.getTitle());
    }

}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/main/widget/adapter/RecommendActivityAdapter.kt
package com.bilibilirecommend.main.widget.adapter


import com.bilibilirecommend.R
import com.bilibilirecommend.base.BaseListRecyclerAdapter
import com.bilibilirecommend.base.SuperViewHolder
import com.bilibilirecommend.main.model.RecommendModel
import com.bilibilirecommend.utils.ImageLoaderUtils

class RecommendActivityAdapter(mDatas: MutableList<RecommendModel.ResultBean.BodyBean>) :
        BaseListRecyclerAdapter<RecommendModel.ResultBean.BodyBean>(mDatas) {

    override val layoutId: Int = R.layout.item_recommend_activity_item

    override fun onBind(viewHolder: SuperViewHolder, position: Int, mDatas: RecommendModel.ResultBean.BodyBean) {
        ImageLoaderUtils.display(viewHolder.getImageView(R.id.iv_title_page), mDatas.cover)
        viewHolder.setTextView(R.id.tv_title, mDatas.title)
    }

}
